km52/Stetsenko_Vladyslav/laboratory1.sql

```diff
@@ -1,2 +1,113 @@
--- LABORATORY WORK 1
--- BY Stetsenko_Vladyslav
+/*---------------------------------------------------------------------------
+1.Створити схему даних з назвою – прізвище студента, у якій користувач може: 
+видаляти дані з таблиці
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+CREATE USER stetsenko IDENTIFIED BY password;
+GRANT DELETE FROM ANY TABLE to stetsenko;
+
+
+
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+2. Створити таблиці, у яких визначити поля та типи. Головні та зовнішні ключі 
+створювати окремо від таблиць використовуючи команди ALTER TABLE. 
+Людина має лікарняну картку, що містить записи про історію хвороби.
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+CREATE TABLE human (surname VARCHAR2(10));
+ALTER TABLE human  
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+  
+/* --------------------------------------------------------------------------- 
+3. Надати додаткові права користувачеві (створеному у пункті № 1) для створення таблиць, 
+внесення даних у таблиці та виконання вибірок використовуючи команду ALTER/GRANT. 
+Згенерувати базу даних використовуючи код з теки OracleScript та виконати запити: 
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+--GRANT CREATE FOR ANY TABLE to stetsenko;
+GRANT INSERT FOR ANY TABLE to stetsenko;
+GRANT SELECT FOR ANY TABLE to stetsenko;
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+3.a. 
+Як звуть постачальника, що продав найдорожчий товар.
+Виконати завдання в Алгебрі Кодда. 
+4 бали
+---------------------------------------------------------------------------*/
+
+--Код відповідь:
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+3.b. 
+Як звуть покупця з найдовшим іменем – поле назвати long_name.
+Виконати завдання в SQL. 
+4 бали
+
+---------------------------------------------------------------------------*/
+
+--Код відповідь:
+SELECT MAX(name_length) FROM (
+  SELECT LENGTH(CUST_NAME) as name_length, CUSTOMERS.CUST_NAME FROM CUSTOMERS
+);
+
+
+
+
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+c. 
+Вивести імена постачальників у нижньому регістрі,назвавши це поле vendor_name, що мають товар і його хтось купив.
+Виконати завдання в SQL. 
+4 бали
+
```

Comment:
RENAME( PROJECT DISTINCT VENDORS TIMES PRODUCTS TIMES ORDERITEMS {LOWER(VENDORS.VEND_ID)} WHERE VENDORS.VEND_ID = PRODUCTS.VEND_ID AND PRODUCTS.PROD_ID = ORDERITEMS.PROD_ID) {VENDORS.VEND_NAME => "vendor_name};